### PR TITLE
Add: `uniqueDeviceIdentifier` parameter to the `configure()` methods

### PR DESCRIPTION
**Pull request by Rolf-Smit**
Currently Parley uses the `Settings.Secure.ANDROID_ID` as unique device identifier, but that comes with a few problems:

- This identifier is not recommended to be used by Google
- This identifier can behave differently on different Android versions
- On more recent Android versions this identifier only resets if the whole device is reset, not on app reinstall.
- Without the ability to change the device ID that Parley uses, it is not possible to fully separate 2 chat users without resetting the device. For example: in an app I'm working on a user can login but also logout, allowing another user to use the app. Both users will now use the same device ID, which is less then ideal.

This pull-request adds a parameter to the `configure()` methods to manually override the device ID that Parley uses, it also adds a warning (Logcat) if no manual ID is provided and deprecates the old `configure()` methods.

### CODE REVIEW

**Comment by Rolf-Smit** [comment]
Update:

Instead of a setter method, the unique device identifier can now be provided through the `configure()` methods.

**Comment by alexkok** [comment]
Thanks @Rolf-Smit ! Looks good -- Requested a few changes.

As for the deprecated note: We will remain the default `ANDROID_ID` usage in the upcoming update for backwards compatible reasons, but will likely update this in a later version by removing the `ANDROID_ID` usage fully. However, for the upcoming update the warning will clearly indicate it at least 👍

**Review by alexkok — CHANGES_REQUESTED** [review]

File: parley/src/main/java/nu/parley/android/Parley.java [hunk 1]
```diff
@@ -85,8 +85,10 @@ public static Parley getInstance() {
      * Convenience for configure(context, secret, {@link EmptyParleyCallback}).
      *
      * @see #configure(Context, String, ParleyCallback)
+     * @deprecated please use the new {@link #configure(Context, String, String)} method that allows setting a unique device identifier.
      */
     @SuppressWarnings("unused")
+    @Deprecated
```
[the remark this answers is not in the document]
Comment (Rolf-Smit) [reply parley/src/main/java/nu/parley/android/Parley.java]: Done.

File: parley/src/main/java/nu/parley/android/Parley.java [hunk 2]
```diff
@@ -97,10 +99,51 @@ public static void configure(final Context context, final String secret) {
      * @param context  Context of the application.
      * @param secret   Application secret of your Parley instance.
      * @param callback {@link ParleyCallback} indicating the result of configuring.
+     *
+     * @deprecated please use the new {@link #configure(Context, String, String, ParleyCallback)} method that allows setting a unique device identifier.
      */
     @SuppressWarnings("WeakerAccess")
+    @Deprecated
```
[the remark this answers is not in the document]
Comment (Rolf-Smit) [reply parley/src/main/java/nu/parley/android/Parley.java]: Done.

File: parley/src/main/java/nu/parley/android/Parley.java [hunk 3]
```diff
@@ -97,10 +99,51 @@ public static void configure(final Context context, final String secret) {
...
+     * <p>
+     * This configure method allows for setting a unique device identifier. If non is provided (by
+     * calling the deprecated configure method), Parley will default to Settings.Secure.ANDROID_ID.
+     * The unique device ID provided to this configure method is not stored by Parley and only kept
+     * for the current instance of Parley. Client applications are responsible for storing it and
+     * providing parley with the same ID. This gives client applications the flexibility to change
+     * the ID if required (for example when another user is logged-in to the app).
+     * <p>
+     * <p>
+     * It is highly recommended to use this method over the deprecated
```
[the remark this answers is not in the document]
Comment (Rolf-Smit) [reply parley/src/main/java/nu/parley/android/Parley.java:133]: Done.

**Comment by Rolf-Smit** [comment]
I've implemented the requested changes :)

**Review by alexkok — APPROVED** [review]

**Comment by alexkok** [comment]
Thanks 👍

**Pull request merged**